tests/recipes/test_hostpython3.py

```diff
@@ -0,0 +1,164 @@
+import unittest
+
+from os.path import join
+from unittest import mock
+
+from pythonforandroid.recipes.hostpython3 import (
+    HOSTPYTHON_VERSION_UNSET_MESSAGE, SETUP_DIST_NOT_FIND_MESSAGE,
+)
+from pythonforandroid.util import BuildInterruptingException
+from tests.recipes.recipe_lib_test import RecipeCtx
+
+
+class TestHostPython3Recipe(RecipeCtx, unittest.TestCase):
+    """
+    TestCase for recipe :mod:`~pythonforandroid.recipes.hostpython3`
+    """
+    recipe_name = "hostpython3"
+
+    def test_property__exe_name_no_version(self):
+        hostpython_version = self.recipe.version
+        self.recipe._version = None
+        with self.assertRaises(BuildInterruptingException) as e:
+            py_exe = self.recipe._exe_name  # noqa: F841
+        self.assertEqual(e.exception.args[0], HOSTPYTHON_VERSION_UNSET_MESSAGE)
+        # restore recipe's version or we will get failures with other test,
+        # since we share `self.recipe with all the tests of the class
+        self.recipe._version = hostpython_version
```

Comment:
I think this is probably fine, but usually a safe way to handle theses things is through context managers.
e.g. https://github.com/kivy/python-for-android/pull/1948
I would say let's keep it simple as you did, but just wanted to share the use of this old PR